tests/python_package_test/test_arrow.py

```diff
@@ -20,6 +20,10 @@
 else:
     import pyarrow as pa  # type: ignore
 
+    assert (
+        lgb.compat.PYARROW_INSTALLED is True
+    ), "'pyarrow' and its dependencies must be installed to run the arrow tests"
```

Comment:
I found myself in a situation where `pyarrow` was installed but `cffi` was not.

That cause this import to fail:

https://github.com/microsoft/LightGBM/blob/e0cda880fc74ca6d1b7d6cb425a24e3a69764bb1/python-package/lightgbm/compat.py#L212

which led to all of the `pyarrow` imports being defined with the placeholders like this:

https://github.com/microsoft/LightGBM/blob/e0cda880fc74ca6d1b7d6cb425a24e3a69764bb1/python-package/lightgbm/compat.py#L227-L231

which led to every Arrow test failing in a hard-to-understand way.

```shell
pytest tests/python_package_test/test_arrow.py
```

```text
==== short test summary info ====
FAILED tests/python_package_test/test_arrow.py::test_dataset_construct_fuzzy[<lambda>-dataset_params0] - AssertionError: assert False
... 
FAILED tests/python_package_test/test_arrow.py::test_predict_ranking - TypeError: Wrong type(ChunkedArray) for label.
==== 139 failed in 11.77s ====
```

As of this change, it'll fail with a slightly clearer error message (and earlier, without running all the tests).

```text
E   AssertionError: 'pyarrow' and its dependencies must be installed to run the arrow tests
```